test/introspection/ERC165Checker.test.js

```diff
@@ -0,0 +1,137 @@
+    });
+
+    it('does not support mock interface via supportsInterface', async function () {
+      const supported = await this.mock.supportsInterface(this.target.address, DUMMY_ID);
+      supported.should.equal(false);
+    });
+
+    it('does not support mock interface via supportsInterfaces', async function () {
+      const supported = await this.mock.supportsInterfaces(this.target.address, [DUMMY_ID]);
+      supported.should.equal(false);
+    });
+  });
+
+  context('ERC165 and single interface supported', function () {
+    beforeEach(async function () {
+      this.target = await ERC165InterfacesSupported.new([DUMMY_ID]);
+    });
+
+    it('supports ERC165', async function () {
+      const supported = await this.mock.supportsERC165(this.target.address);
+      supported.should.equal(true);
+    });
+
+    it('supports mock interface via supportsInterface', async function () {
+      const supported = await this.mock.supportsInterface(this.target.address, DUMMY_ID);
+      supported.should.equal(true);
+    });
+
+    it('supports mock interface via supportsInterfaces', async function () {
+      const supported = await this.mock.supportsInterfaces(this.target.address, [DUMMY_ID]);
+      supported.should.equal(true);
+    });
+  });
+
+  context('ERC165 and many interfaces supported', function () {
+    beforeEach(async function () {
+      this.supportedInterfaces = [DUMMY_ID, DUMMY_ID_2, DUMMY_ID_3];
+      this.target = await ERC165InterfacesSupported.new(this.supportedInterfaces);
+    });
+
+    it('supports ERC165', async function () {
+      const supported = await this.mock.supportsERC165(this.target.address);
+      supported.should.equal(true);
+    });
+
+    it('supports each interfaceId via supportsInterface', async function () {
+      for (const interfaceId of this.supportedInterfaces) {
+        const supported = await this.mock.supportsInterface(this.target.address, interfaceId);
+        supported.should.equal(true);
+      };
+    });
+
+    it('supports all interfaceIds via supportsInterfaces', async function () {
+      const supported = await this.mock.supportsInterfaces(this.target.address, this.supportedInterfaces);
+      supported.should.equal(true);
+    });
+
+    it('supports none of the interfaces queried via supportsInterfaces', async function () {
+      const interfaceIdsToTest = [DUMMY_UNSUPPORTED_ID, DUMMY_UNSUPPORTED_ID_2];
+
+      const supported = await this.mock.supportsInterfaces(this.target.address, interfaceIdsToTest);
+      supported.should.equal(false);
+    });
+
+    it('supports not all of the interfaces queried via supportsInterfaces', async function () {
+      const interfaceIdsToTest = [...this.supportedInterfaces, DUMMY_UNSUPPORTED_ID];
+
+      const supported = await this.mock.supportsInterfaces(this.target.address, interfaceIdsToTest);
+      supported.should.equal(false);
+    });
+  });
+
+  context('account address does not support ERC165', function () {
```

Comment:
Aren't these tests repeats of `context('ERC165 not supported')`?

Reply:
but specifically for EOA accounts, not a contract deployed to the chain

Reply:
right, I guess those are different enough to make this a meaningful test, thanks!